application_manager/plugins/spark_mesos/plugin.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2017 UFCG-LSD and UPV-GRyCAP.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from application_manager.plugins import base
+from application_manager.service import api
+from application_manager.utils import monitor
+from application_manager.utils import scaler
+from application_manager.utils.logger import Log, configure_logging
+
+from uuid import uuid4
+
+import json
+import paramiko
+import time
+
+plugin_log = Log("Spark-Mesos_Plugin", "logs/mesos_plugin.log")
+configure_logging()
+
+
+class SparkMesosProvider(base.PluginInterface):
+
+    def __init__(self):
+        self.get_frameworks_url = "%s:%s" % (api.mesos_url,
+                                             api.mesos_port)
+        self.app_id = "app-spark-mesos-" + str(uuid4())[:8]
+
+
+    def get_title(self):
+        return 'Spark-Mesos on Open Nebula plugin for BigSea framework'
+
+    def get_description(self):
+        return 'It runs an spark application on a Mesos cluster'
+
+    def to_dict(self):
+        return {
+            'name': self.name,
+            'title': self.get_title(),
+            'description': self.get_description(),
+        }
+
+    def execute(self, data):
```

Comment:
Is it possible to split this method in submethods like "get_framework", "get_execution_vm_ids", "get_execution_vm_ips", "start_application"?

Reply:
Yes. This will be done.